Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,33 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      # explicitly use our npm and node-gyp executables instead of the user
+      # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
+      ENV.prepend_path "PATH", Formula["node"].opt_libexec/"npm/bin"
+      # explicitly set npm's cache path to HOMEBREW_CACHE/npm_cache to fix
+      # issues caused by overriding $HOME (long build times, high disk usage)
+      # https://github.com/Homebrew/brew/pull/37#issuecomment-208840366
+      npmrc = Pathname.new("#{ENV["HOME"]}/.npmrc")
+      npmrc.write npm_cache_config unless npmrc.exist?
+    end
+
+    def self.std_npm_install_args(libexec)
+      setup_npm_environment
+      # tell npm to not install .brew_home by adding it to the .npmignore file
+      # (or creating a new one if no .npmignore file already exists)
+      open(".npmignore", "a") { |f| f.write( "\n.brew_home\n") }
+      # npm install args for global style module format installed into libexec
+      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+    end
```

Comment:
So if I call `std_npm_install_args` multiple times, I will end up with duplicated `Formula["node"].opt_libexec/"npm/bin"` in `PATH`, as well as `.npmignore` file duplicated entries.

Ultimately, it feels a little weird to have  `std_npm_install_args` and `local_npm_install_args` which supports to just return an argument list, to change any file system.

My suggest solutions are:
 * (a) create a separate method to setup environment. and call this method explicitly in formula.
 * (b) Or, make sure setup is only run once.